third-party/gasnet/Makefile

```diff
@@ -45,6 +45,13 @@ CHPL_GASNET_CFG_OPTIONS += -with-mpi-cc=mpcc
 endif
 
 ifneq (, $(filter $(CHPL_MAKE_TARGET_PLATFORM),cray-xe cray-xc cray-xk))
+
+ifneq ($(CHPL_MAKE_COMM_SUBSTRATE),mpi)
```

Comment:
There can be other substrates like smp/ofi. I would just put this in the checks for aries/gemini below so we only throw it for aries/gem